### PR TITLE
Add support for seeding RNG used for random sampling (#59).

### PR DESCRIPTION
**Pull request by lebedov**
This PR adds support for specifying a seed for the RNG used to create a random ref set when computing the gap statistic; this effectively permits one to make computing the gap statistic reproducible per the request in #59 if one uses a clustering algorithm that is either deterministic or uses an RNG that can be seeded separately. Here is a snippet of code that verifies that the modified code produces the same gap statistics when invoked using the same RNG seed:

```python
from gap_statistic import OptimalK
import numpy as np
from sklearn.cluster import AgglomerativeClustering
from sklearn.datasets import load_digits
from sklearn.neighbors import NearestCentroid

def agglomerative_clustering_func(x, k):
    cluster = AgglomerativeClustering(n_clusters=k)
    cluster_list = cluster.fit_predict(x)
    clf = NearestCentroid()
    clf.fit(x, cluster_list)
    return clf.centroids_, clf.classes_

if __name__ == '__main__':
    data = load_digits()

    s = set()
    for i in range(10):
        optimalK = OptimalK(random_state=0,
                            clusterer=agglomerative_clustering_func,
                            parallel_backend='na')
        n_clusters = optimalK(data.data, cluster_array=np.arange(2, 20))
        print(optimalK.gap_df.head())
        s.add(n_clusters)
    print('number of clusters: ', s)
```

Given that the kmeans implementations in scipy and sklearn seemingly don't currently provide access to the RNGs used for kmeans initialization (when not explicitly specifying centroids), there seemed less reason to do the same for the Rust-based implementation included in gap-stat; as such, I did not modify the Rust code.

### CODE REVIEW

**Comment by milesgranger** [comment]
Thanks for this! I'm not ignoring it, only on vacation at the moment. Will review/merge/release when I'm back. Thanks again! 👍

File: Cargo.lock [hunk 1]
```diff
@@ -220,7 +222,7 @@ checksum = "a06f77d526c1a601b7c4cdd98f54b5eaabffc14d5f2f0296febdc7f357c6d3ba"
 
 [[package]]
 name = "gapstat-rs"
-version = "2.0.0-rc2"
+version = "2.0.1"
```
Comment (milesgranger) [inline Cargo.lock:225]: I'm a bit confused why changes in `Cargo.lock`? Not relevant to the changes in python files.

**Comment by milesgranger** [comment]
I suspect the CI might be sticky, currently failing and ought to be updated to at least 3.9 IMO. Likely other things may be wrong after since this repo isn't very active and thus not run very often.

I'm fine if we do that in a follow up PR.

File: Cargo.lock (hunk 1 above)
Comment (lebedov) [reply Cargo.lock]: Ah, I think I updated that file before I decided that setting the rng seed for k-means would go beyond what is currently possible with sklearn. Reverted.

**Review by milesgranger — APPROVED** [review]
Thanks!

Again, the CI will need fixing so not bothering to run it now. It at least needs updating to a more recent python version. I'll publish a source distribution with these changes soon though.

https://github.com/milesgranger/gap_statistic/issues/62

**Comment by milesgranger** [comment]
https://github.com/milesgranger/gap_statistic/releases/tag/v2.0.3

https://pypi.org/project/gap-stat/#files

**Comment by lebedov** [comment]
Thanks! Re python version, I initially used typing syntax that I subsequently discovered wasn't supported until py 3.10, so I removed that change from the PR.
